clients/fides-js/src/fides-tcf.ts

```diff
@@ -80,29 +103,110 @@ declare global {
 // eslint-disable-next-line no-underscore-dangle,@typescript-eslint/naming-convention
 let _Fides: Fides;
 
+/** Helper function to determine the initial value of a TCF object */
+const getInitialPreference = (
+  tcfObject:
+    | TCFFeatureRecord
+    | TCFVendorConsentRecord
+    | TCFVendorLegitimateInterestsRecord
+): UserConsentPreference => {
+  if (tcfObject.current_preference) {
+    return tcfObject.current_preference;
+  }
+  return tcfObject.default_preference ?? UserConsentPreference.OPT_OUT;
+};
+
 const updateCookie = async (
   oldCookie: FidesCookie,
-  experience: PrivacyExperience
+  experience: PrivacyExperience,
+  debug?: boolean,
+  tc_str_override_set?: boolean
 ): Promise<FidesCookie> => {
-  // First check if the user has never consented before
+  // ignore server-side prefs if either user has no prefs to override, or TC str override is set
+  if (tc_str_override_set) {
+    return oldCookie;
+  }
   if (!hasSavedTcfPreferences(experience)) {
     return { ...oldCookie, tc_string: "" };
   }
 
-  // Usually at this point, we'd look at the Experience from the backend and update
-  // the user's browser cookie to match the preferences in the Experience. However,
-  // TCF requires pre-fetching an experience. A prefetch'd experience will never have user
-  // specific consents. We rely on the cookie to fill those in. Therefore, we do nothing
-  // here, since the cookie is the source of truth, not the backend Experience.
-  return oldCookie;
+  const tcSavePrefs: TcfSavePreferences = {};
+  const enabledIds: any | EnabledIds = {
+    purposesConsent: [],
+    purposesLegint: [],
+    specialPurposes: [],
+    features: [],
+    specialFeatures: [],
+    vendorsConsent: [],
+    vendorsLegint: [],
+  };
+
+  TCF_KEY_MAP.forEach(({ experienceKey, cookieKey, enabledIdsKey }) => {
+    if (experienceKey) {
```

Comment:
`experienceKey` always exists so I don't think you need the `if` here

Reply:
This was more for a possible scenario in the future where the `TCF_KEY_MAP` would contain an item that exists in the cookie keys but we don't track in an experience. Is this something we should anticipate you think?

Reply:
hmm hard to say! I hope not... I'd prefer to keep things matching with how they are now rather than building towards a potential future, but in this case it's not a huge thing